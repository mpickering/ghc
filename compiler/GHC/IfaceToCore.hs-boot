module GHC.IfaceToCore where

import GhcPrelude
import GHC.Iface.Syntax ( IfaceDecl, IfaceClsInst, IfaceFamInst, IfaceRule
                        , IfaceAnnotation, IfaceCompleteMatch, IfaceExpr, IfaceType )
import GHC.Core.TyCo.Rep   ( TyThing, Type )
import TcRnTypes           ( IfL )
import GHC.Core.InstEnv    ( ClsInst )
import GHC.Core.FamInstEnv ( FamInst )
import GHC.Core         ( CoreRule, CoreExpr )
import GHC.Driver.Types ( CompleteMatch )
import Annotations      ( Annotation )
import Name (Name)

tcIfaceDecl         :: Bool -> IfaceDecl -> IfL TyThing
tcIfaceRules        :: Bool -> [IfaceRule] -> IfL [CoreRule]
tcIfaceInst         :: IfaceClsInst -> IfL ClsInst
tcIfaceFamInst      :: IfaceFamInst -> IfL FamInst
tcIfaceAnnotations  :: [IfaceAnnotation] -> IfL [Annotation]
tcIfaceCompleteSigs :: [IfaceCompleteMatch] -> IfL [CompleteMatch]
tcIfaceGlobal :: Name -> IfL TyThing
tcIfaceExpr :: IfaceExpr -> IfL CoreExpr
tcIfaceType :: IfaceType -> IfL Type
