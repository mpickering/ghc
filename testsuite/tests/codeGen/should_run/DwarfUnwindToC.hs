import GHC.ExecutionStack

-- | Trim object file names
cleanUpBacktrace :: String -> String
cleanUpBacktrace = unlines . map trimLine . lines
  where
    trimLine (' ':'i':'n':' ':_) = ""
    trimLine (x:xs)              = x : trimLine xs
    trimLine []                  = []

test :: Int -> IO ()
test 0 = return ()
test i = do
    print i
    showStackTrace >>= putStrLn . cleanUpBacktrace
    test (i-1)
    return ()

main = do
        test 3
        print "Hello"
