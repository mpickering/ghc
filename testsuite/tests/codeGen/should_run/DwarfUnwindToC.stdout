3
Stack trace:
    set_initial_registers (rts/Libdw.c:278.0)
    dwfl_thread_getframes
    
    dwfl_getthreads
    dwfl_getthread_frames
    libdw_get_backtrace (rts/Libdw.c:249.0)
    base_GHCziExecutionStackziInternal_collectStackTrace1_info (libraries/base/GHC/ExecutionStack/Internal.hsc:194.1)
    base_GHCziExecutionStack_showStackTrace1_info (libraries/base/GHC/ExecutionStack.hs:47.1)
    base_GHCziBase_bindIO1_info (libraries/base/GHC/Base.hs:1081.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    stg_catch_frame_info (rts/Exception.cmm:370.1)
    stg_stop_thread_info (rts/StgStartup.cmm:42.1)

2
Stack trace:
    set_initial_registers (rts/Libdw.c:278.0)
    dwfl_thread_getframes
    
    dwfl_getthreads
    dwfl_getthread_frames
    libdw_get_backtrace (rts/Libdw.c:249.0)
    base_GHCziExecutionStackziInternal_collectStackTrace1_info (libraries/base/GHC/ExecutionStack/Internal.hsc:194.1)
    base_GHCziExecutionStack_showStackTrace1_info (libraries/base/GHC/ExecutionStack.hs:47.1)
    base_GHCziBase_bindIO1_info (libraries/base/GHC/Base.hs:1081.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    stg_catch_frame_info (rts/Exception.cmm:370.1)
    stg_stop_thread_info (rts/StgStartup.cmm:42.1)

1
Stack trace:
    set_initial_registers (rts/Libdw.c:278.0)
    dwfl_thread_getframes
    
    dwfl_getthreads
    dwfl_getthread_frames
    libdw_get_backtrace (rts/Libdw.c:249.0)
    base_GHCziExecutionStackziInternal_collectStackTrace1_info (libraries/base/GHC/ExecutionStack/Internal.hsc:194.1)
    base_GHCziExecutionStack_showStackTrace1_info (libraries/base/GHC/ExecutionStack.hs:47.1)
    base_GHCziBase_bindIO1_info (libraries/base/GHC/Base.hs:1081.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    base_GHCziBase_thenIO1_info (libraries/base/GHC/Base.hs:1084.1)
    stg_catch_frame_info (rts/Exception.cmm:370.1)
    stg_stop_thread_info (rts/StgStartup.cmm:42.1)

"Hello"
